src/System.Memory/src/System/Buffers/StandardFormat.cs

```diff
@@ -111,6 +121,47 @@ public static StandardFormat Parse(string format)
         public bool Equals(StandardFormat other) => _format == other._format && _precision == other._precision;
 
         /// <summary>
+        /// Returns the format in classic .NET format.
+        /// </summary>
+        public override string ToString()
+        {
+            unsafe
+            {
+                const int MaxLength = 4;
+                char* pBuffer = stackalloc char[MaxLength];
+
+                int dstIndex = 0;
+                char symbol = Symbol;
+                if (symbol != default)
+                {
+                    pBuffer[dstIndex++] = symbol;
+
+                    byte precision = Precision;
+                    if (precision != NoPrecision)
+                    {
+                        if (precision >= 100)
```

Comment:
Is this considered valid if MaxPrecision is 99? I thought MaxLength would be 3, not 4.

Reply:
`ToString()` cannot throw exceptions and there's no way to prevent a mischievous app from hand-crafting a struct with illegal values in it. So yes, it has to be robust in this situation.

Reply:
I'm not understanding this argument.  If you corrupt the data structure with unsafe code or the equivalent, then yeah, this might give you an invalid answer, e.g. 23 instead of 123.  That's fine.  It won't corrupt anything, it'll just give an invalid answer for invalid input.